src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/OrderEndpoints.scala

```diff
@@ -6,12 +6,19 @@ import io.circe.syntax._
 import org.http4s._
 import org.http4s.circe._
 import org.http4s.dsl.Http4sDsl
-import scala.language.higherKinds
 
+import scala.language.higherKinds
 import io.github.pauljamescleary.petstore.domain.OrderNotFoundError
+import io.github.pauljamescleary.petstore.domain.authentication.Auth
 import io.github.pauljamescleary.petstore.domain.orders.{Order, OrderService}
+import io.github.pauljamescleary.petstore.domain.users.User
+import tsec.authentication.{AugmentedJWT, SecuredRequest, SecuredRequestHandler, TSecAuthService, asAuthed}
+import tsec.jwt.algorithms.JWTMacAlgo
+
+class OrderEndpoints[F[_]: Effect, Auth: JWTMacAlgo] extends Http4sDsl[F] {
 
-class OrderEndpoints[F[_]: Effect] extends Http4sDsl[F] {
+  type AuthService = TSecAuthService[User, AugmentedJWT[Auth, Long], F]
+  type AuthEndpoint = PartialFunction[SecuredRequest[F, User, AugmentedJWT[Auth, Long]], F[Response[F]]]
```

Comment:
this is really nice.  Seem to repeat these type aliases in all the endpoints, can we have these reused across all endpoints?

Reply:
I've already tried to export those type aliases in their own object but it didn't work. I will give a second try because it's a low hanging fruit.

Reply:
Probably a package object for the endpoint package would work

Reply:
I see, if you factor out the type aliases, you would still have to parameterize just by the `Auth` type, which I think is fine.